blocks/site-details/site-details.js

```diff
@@ -599,6 +599,151 @@ const createDocsEl = (html) => `
+        <p class="warning">Keep in mind, any changes made on the options and theme pages after an update will <strong>also</strong> be reverted! <strong>This action cannot be undone!</strong></p>
+        `;
+      const revertForm = document.createElement('form');
+      revertForm.id = 'revert-form';
+      revertForm.innerHTML = `
+        <form id="revert-form">
+          <ul class="applied-update-list">
+            ${updateList.map((update) => `<li><label><input required type="radio" name="update" data-version="${update.version}" value="${update.sha}"><span>Version: <strong>${update.version}</strong></span><span>Updated on: <strong>${new Date(update.date).toLocaleString(undefined, {
+    year: 'numeric', month: 'numeric', day: 'numeric', hour: 'numeric', minute: 'numeric',
+  })}</strong></span></label></li>`).join('')}
+          </ul>
+        </form>
+      `;
+      dialogContent.append(revertForm);
+
+      const revertButton = document.createElement('button');
+      revertButton.classList.add('button', 'action', 'secondary', 'update-button');
+      revertButton.setAttribute('form', 'revert-form');
+      revertButton.type = 'submit';
+      revertButton.innerText = 'Undo Update';
+      revertButton.disabled = true;
+      revertUpdateDialog.renderDialog(dialogContent, [revertButton]);
+
+      let currentSelectedUpdate = null;
+      revertForm.onchange = (event) => {
+        currentSelectedUpdate = event.target.dataset.version;
+        if (currentSelectedUpdate) {
+          revertButton.disabled = null;
+        }
+      };
+
+      revertForm.onsubmit = async (event) => {
+        event.preventDefault();
+        window?.zaraz?.track('did site update revert', { url: window.location.href });
+
+        if (await window.confirmDialog(`<div><p class="warning">Are you sure you want to revert ${`the ${currentSelectedUpdate}` || 'to before a previous'} update?</p><p><strong>any changes made on the options and theme pages after an update will also be reverted!</strong></p><p class="warning"><strong>This action cannot be undone!</strong></p></div>`)) {
+          revertUpdateDialog.dataset.loadingText = 'Reverting to previous version...';
+          revertUpdateDialog.setLoading(true);
+          const formData = new FormData(revertForm);
+          const revertUpdateResponse = await fetch(`${endpoint}revert/${project.projectSlug}`, {
+            method: 'POST',
+            headers: { ...headers, 'Content-Type': 'application/json' },
+            body: JSON.stringify({ sha: formData.get('update') }),
+          });
+          if (revertUpdateResponse.ok) {
+            revertUpdateDialog.renderDialog('<h3 class="centered-info">Project reverted successfully!</h3>');
+            // replace update button/up-to-date message, check for updates after 6sec
+            updateInfoDiv.innerHTML = '<h3>Project reverted successfully!</h3>';
+            setTimeout(() => rerenderUpdatesSection(updateInfoDiv, { project, headers }), 15000);
```

Comment:
Mhh looks suspicious :) 15s ??

Reply:
I was getting the template version from the url in EDS. This takes some time to update, but I'll change it to get the contents from github. It just saved us some API calls, but I guess it's better to respond quickly

Reply:
This has been changed, along with the backend to always have the most accurate templateInfo be sent